File: 0787. Cheapest Flights Within K Stops/Solution0787.cpp
class Solution0787 {
public:
    int findCheapestPrice(int n, vector<vector<int>>& flights, int src, int dst, int k) {
        vector<int> dp(n, INT_MAX/2);
        dp[src] = 0;
        for (int i = 0; i < k + 1; i++) {
            vector<int> temp(n);
            copy(dp.begin(), dp.end(), temp.begin());
            for (auto flight : flights) {
                dp[flight[1]] = min(temp[flight[0]] + flight[2], dp[flight[1]]);
            }
        }
        if (dp[dst] != INT_MAX/2) {
            return dp[dst];
        }
        return -1;
    }
};

File: 0787. Cheapest Flights Within K Stops/Solution0787.go
import "math"

func findCheapestPrice(n int, flights [][]int, src int, dst int, k int) int {
	prev := []int{}
	now := make([]int, n)
	for j := range now {
		now[j] = math.MaxInt32
	}
	now[src] = 0
	for i := 0; i < k+1; i++ {
		prev = now
		now = append([]int{}, prev...)
		for _, flight := range flights {
			now[flight[1]] = min(now[flight[1]], prev[flight[0]]+flight[2])
		}
	}
	if now[dst] == math.MaxInt32 {
		return -1
	}
	return now[dst]
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}